internal/ethereum/exec_query.go

```diff
@@ -70,39 +78,42 @@ func (c *ethConnector) QueryInvoke(ctx context.Context, req *ffcapi.QueryInvokeR
 
 }
 
-func (c *ethConnector) callTransaction(ctx context.Context, tx *ethsigner.Transaction, method *abi.Entry) (*fftypes.JSONAny, ffcapi.ErrorReason, error) {
+func (c *ethConnector) callTransaction(ctx context.Context, tx *ethsigner.Transaction, method *abi.Entry, errors []*abi.Entry) (*fftypes.JSONAny, ffcapi.ErrorReason, error) {
 
 	// Do the raw call
 	var outputData ethtypes.HexBytes0xPrefix
-	err := c.backend.CallRPC(ctx, &outputData, "eth_call", tx, "latest")
-	if err != nil {
-		return nil, mapError(callRPCMethods, err), err
+	rpcErr := c.backend.CallRPC(ctx, &outputData, "eth_call", tx, "latest")
+	if rpcErr != nil {
+		// some Ethereum implementations (eg. geth 1.10) returns the revert data inside
+		// the "error" object rather than the "result" object in the response
+		if rpcErr.Data != "" {
+			var revertData ethtypes.HexBytes0xPrefix
+			e1 := json.Unmarshal(rpcErr.Data.Bytes(), &revertData)
+			if e1 != nil {
+				return nil, mapError(callRPCMethods, e1), e1
```

Comment:
I'm not sure we want to lose and override the original error here @jimthematrix 
Maybe better to log the full `rpcError` object and then just return `rpcError.Error()`?